main.go

```diff
@@ -325,6 +331,57 @@ func (s *service) openNewIssues(issueMap map[string]*github.Issue, comments []*t
 	log.Printf("Created new issues. count=%v", count)
 }
 
+func (s *service) assignNewIssues(issueMap map[string]*github.Issue, assigneeMap *map[string]string) {
+	for title, assignee := range *assigneeMap {
+		issue := issueMap[title]
+		issueNumber := issue.GetNumber()
+		req := &github.IssueRequest{
+			Assignees: &[]string{assignee},
+		}
+		_, _, err := s.client.Issues.Edit(s.ctx, s.env.owner, s.env.repo, issueNumber, req)
+		if err != nil {
+			log.Printf("Error while assigning %v to issue %v. err=%v", assignee, issueNumber, err)
+		} else {
+			log.Printf("Successfully assigned %v to issue %v.", assignee, issueNumber)
+		}
+	}
+}
+
+func (s *service) getAssigneeFromCommitHash(commitHash string, title string, assigneeMap *map[string]string, successCount *int, assigneeMapMux sync.Locker, wg *sync.WaitGroup) {
+	defer wg.Done()
+	commit, _, err := s.client.Repositories.GetCommit(s.ctx, s.env.owner, s.env.repo, commitHash, &github.ListOptions{})
+	if err != nil {
+		log.Printf("Error while getting commit from commit hash. err=%v", err)
+	} else {
+		assigneeMapMux.Lock()
+		defer assigneeMapMux.Unlock()
+		(*assigneeMap)[title] = *commit.Author.Login
+		*successCount++
+	}
+}
+
+func (s *service) getAssigneesForNewIssues(issueMap map[string]*github.Issue, comments []*tdglib.ToDoComment, assigneeMap *map[string]string) {
+	defer s.wg.Done()
+
+	var wg sync.WaitGroup
```

Comment:
What exactly is the reason we need a new wait group and cannot reuse the one from the `service`? We're still waiting for that wait group in the `main()` before we return from here.

Reply:
I did this because I wanted to output the number of issues for which we successfully got assignees. So, for that, we want all the threads to have completed. If we use the other existing wg, then we would have to get this info only after the wait command for that, but that is outside this function and the code would be a bit messy.